src/platform_impl/linux/wayland/window/state.rs

```diff
@@ -197,6 +200,7 @@ impl WindowState {
             resizable: true,
             scale_factor: 1.,
             shm: winit_state.shm.wl_shm().clone(),
+            pool: SlotPool::new(4, &winit_state.shm).unwrap(),
```

Comment:
The pool must be shared for all the windows and instantiated if and only if we actually try to set the custom cursor.

Reply:
Now it is shared but I was unable to instantiate it lazily. The window doesn't have access to the actual `Shm` that is needed to create the pool. It has access to `WlShm` but that isn't enough. 

Also looking at SCTK code doesn't help because they do it with "raw" memfds and shms kind of like I did before [link](https://github.com/Smithay/wayland-rs/blob/fdd88c4fa2f5c5abaa3044f62c4498f5f7b7dbd4/wayland-cursor/src/lib.rs#L125-L164). SlotPool seems more convenient though so we should continue to use that. It also seems to be able to actually reuse memory while `wayland-cursor` doesn't care about that because there are so few named cursor types in existence.